sdk/core/azure-core-tracing-opentelemetry/azure/core/tracing/ext/opentelemetry_span/__init__.py

```diff
@@ -0,0 +1,250 @@
+        configure the azure exporter else will just create a new tracer.
+
+        :param span: The OpenTelemetry span to wrap
+        :type span: :class: OpenTelemetry.trace.Span
+        :param name: The name of the OpenTelemetry span to create if a new span is needed
+        :type name: str
+        """
+        current_tracer = self.get_current_tracer()
+        self._span_instance = span or current_tracer.start_span(name=name)
+        self._current_ctxt_manager = None
+
+    @property
+    def span_instance(self):
+        # type: () -> Span
+        """
+        :return: The OpenTelemetry span that is being wrapped.
+        """
+        return self._span_instance
+
+    def span(self, name="span"):
+        # type: (Optional[str]) -> OpenCensusSpan
+        """
+        Create a child span for the current span and append it to the child spans list in the span instance.
+        :param name: Name of the child span
+        :type name: str
+        :return: The OpenCensusSpan that is wrapping the child span instance
+        """
+        return self.__class__(name=name)
+
+    @property
+    def kind(self):
+        # type: () -> Optional[SpanKind]
+        """Get the span kind of this span."""
+        value = self.span_instance.kind
+        return (
+            SpanKind.CLIENT if value == OpenTelemetrySpanKind.CLIENT else
+            SpanKind.PRODUCER if value == OpenTelemetrySpanKind.PRODUCER else
+            SpanKind.SERVER if value == OpenTelemetrySpanKind.SERVER else
+            SpanKind.CONSUMER if value == OpenTelemetrySpanKind.CONSUMER else
+            SpanKind.INTERNAL if value == OpenTelemetrySpanKind.INTERNAL else
+            None
+        )
+
+
+    @kind.setter
+    def kind(self, value):
+        # type: (SpanKind) -> None
+        """Set the span kind of this span."""
+        kind = (
+            OpenTelemetrySpanKind.CLIENT if value == SpanKind.CLIENT else
+            OpenTelemetrySpanKind.PRODUCER if value == SpanKind.PRODUCER else
+            OpenTelemetrySpanKind.SERVER if value == SpanKind.SERVER else
+            OpenTelemetrySpanKind.CONSUMER if value == SpanKind.CONSUMER else
+            OpenTelemetrySpanKind.INTERNAL if value == SpanKind.INTERNAL else
+            OpenTelemetrySpanKind.INTERNAL if value == SpanKind.UNSPECIFIED else
+            None
+        )
+        if kind is None:
+            raise ValueError("Kind {} is not supported in OpenTelemetry".format(value))
+        self.span_instance.kind = kind
+
+    def __enter__(self):
+        """Start a span."""
+        self.start()
+        self._current_ctxt_manager = self.get_current_tracer().use_span(self._span_instance, end_on_exit=True)
+        self._current_ctxt_manager.__enter__()
+        return self
+
+    def __exit__(self, exception_type, exception_value, traceback):
+        """Finish a span."""
+        if not self._current_ctxt_manager:
```

Comment:
What is the expected semantic of calling `__exit__()` multiple times? Currently, there is nothing guarding that (we are not clearing the `_current_ctxt_manager` instance on `__exit__`. Is it really important for us to guard on entry? 

(performance is important for the tracing plugin, after all)

Reply:
Fix by being less nervous about checking